### PR TITLE
11 login register page form

### LINKED ISSUE #11: Login / Register Page Form
In this issue, there is need to create a form that supports user login and registration

### PR DESCRIPTION
**Pull request by KacperWojdak**
Form for login and register is created and ready

### CODE REVIEW

File: frontend/src/pages/LoginPage.js [hunk 1]
```diff
@@ -1,16 +1,90 @@
...
+        Zaloguj się
+      </button>
+      <p className="text-center mt-4">
+        Nie posiadasz konta?{' '}
+        <a href="#" onClick={toggleForm} className="text-blue-500 underline">
+          Zarejestruj się.
+        </a>
+      </p>
+    </form>
+  )
```
Comment (PiotrFedak) [inline frontend/src/pages/LoginPage.js]: This should be in seperate file in components/Login

File: frontend/src/pages/LoginPage.js [hunk 2]
```diff
@@ -1,16 +1,90 @@
...
+        Zarejestruj się
+      </button>
+      <p className="text-center mt-4">
+        Posiadasz już konto?{' '}
+        <a href="#" onClick={toggleForm} className="text-blue-500 underline">
+          Zaloguj się.
+        </a>
+      </p>
+    </form>
+  )
```
Comment (PiotrFedak) [inline frontend/src/pages/LoginPage.js]: Also this same way as login file, then just import both of them here

**Review by PiotrFedak — CHANGES_REQUESTED** [review]

File: frontend/src/pages/LoginPage.js (hunk 1 above)
Comment (KacperWojdak) [reply frontend/src/pages/LoginPage.js]: ✔️

File: frontend/src/pages/LoginPage.js (hunk 2 above)
Comment (KacperWojdak) [reply frontend/src/pages/LoginPage.js]: ✔️

File: frontend/src/pages/LoginPage.js
Comment (PiotrFedak) [inline frontend/src/pages/LoginPage.js:1]: Can we add some image in the background ? you could use your ai magic tool. 
Or use https://undraw.co/  on this website it is possible to chose what color of svg you want for example your favorite yellow #C58A09

**Review by PiotrFedak — CHANGES_REQUESTED** [review]
I still don't like the overall looks.
For me the color just don't match with each other, you could create a palet of 3-4 color that will match with your yellow, i guess it is something that can be added anytime so you can take your time if you have to.

a13a0132ead49fb38fcce528c7c75521

This is the contrast score of you header

File: frontend/src/pages/LoginPage.js
Comment (KacperWojdak) [reply frontend/src/pages/LoginPage.js:1]: ✔️

File: frontend/src/pages/AuthPage.js
Comment (PiotrFedak) [inline frontend/src/pages/AuthPage.js:1]: 79838cca41904682dfa2c056e09c85d8
It is possible to scroll into nothing, you need to add `overflow-hidden` to block that possibility. Or create a footer that will take the bottom space

**Review by PiotrFedak — CHANGES_REQUESTED** [review]

File: frontend/src/pages/AuthPage.js
Comment (KacperWojdak) [reply frontend/src/pages/AuthPage.js:1]: ✔️

**Review by TomaszBarczycki — COMMENTED** [review]
Apart from everything our PM mentioned, everything seems fine to me

**Review by PiotrFedak — CHANGES_REQUESTED** [review]
There is a lack of axios that is needed for register and login to work.
I would suggest to create a axiosClient.js the way we do in other projects.
Also add "ContextProvider.js" maybe even create a GuestLayout to try if blocking path works.

Of course if you don't like that approach you could try with Auth0 or some other authentication service.

**Review by TomaszBarczycki — CHANGES_REQUESTED** [review]
I'd consider to make the app responsive. When you try to open it with a mobile view, there are a lot of bugs

**Comment by KacperWojdak** [comment]
resp1
resp2
resp3
resp4

For responsive mobile view. Header is changing button text into powrót and making title smaller. On main page a panel is getting narrower and buttons are one by one. From AuthPage images are gone so there is only a login/register form and on map page there is a map and list one by one for now.

File: frontend/src/index.js [hunk 3]
```diff
@@ -2,15 +2,18 @@ import React from 'react'
...
+      <App />
+    </ContextProvider>
   </React.StrictMode>,
 )
 
 // If you want to start measuring performance in your app, pass a function
 // to log results (for example: reportWebVitals(console.log))
 // or send to an analytics endpoint. Learn more: https://bit.ly/CRA-vitals
-reportWebVitals()
+//reportWebVitals()
```
Comment (PiotrFedak) [inline frontend/src/index.js]: Import router here in this file instead of App

File: frontend/src/App.test.js [hunk 4]
```diff
@@ -1,4 +1,5 @@
 import { render, screen } from '@testing-library/react'
+import { expect, test } from '@jest/globals';
 import App from './App'
 
 test('renders learn react link', () => {
```
Comment (PiotrFedak) [inline frontend/src/App.test.js]: Also this, can be remove

File: frontend/src/App.js [hunk 5]
```diff
@@ -1,5 +1,5 @@
 import React from 'react'
-import AppRouter from './router'
+import AppRouter from './router.js'
 
 export default function App() {
   return (
```
Comment (PiotrFedak) [inline frontend/src/App.js]: When you done moving routing to index. then this file can be deleted

**Review by PiotrFedak — COMMENTED** [review]
If u gonna repat some colors like "#DDA15E" then you could add this to tailwind css config file and name it something norumal.

File: frontend/src/index.js (hunk 3 above)
Comment (KacperWojdak) [reply frontend/src/index.js]: ✔️

File: frontend/src/App.test.js (hunk 4 above)
Comment (KacperWojdak) [reply frontend/src/App.test.js]: ✔️

File: frontend/src/App.js (hunk 5 above)
Comment (KacperWojdak) [reply frontend/src/App.js]: ✔️

**Review by PiotrFedak — APPROVED** [review]

**Review by TomaszBarczycki — APPROVED** [review]

**Pull request merged**